new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java

```diff
@@ -0,0 +1,1258 @@
+   *  spec/adminServer/serverStartPolicy set to NEVER.
+   * Make sure that Only the Administration server is stopped. 
+   * Restart the Administration server by patching the resource definition with 
+   *  spec/adminServer/serverStartPolicy set to IF_NEEDED.
+   * Make sure that the Administration server is in RUNNING state. 
+   */
+  @Test
+  @DisplayName("Restart the Administration server with serverStartPolicy")
+  public void testAdminServerRestart() {
+
+    String configServerPodName = domainUid + "-config-cluster-server1";
+    String dynamicServerPodName = domainUid + "-managed-server1";
+    DateTime dynTs = getPodCreationTime(domainNamespace, dynamicServerPodName);
+    DateTime cfgTs = getPodCreationTime(domainNamespace, configServerPodName);
+    StringBuffer patchStr = null;
+    patchStr = new StringBuffer("[{");
+    patchStr.append("\"op\": \"replace\",")
+        .append(" \"path\": \"/spec/adminServer/serverStartPolicy\",")
+        .append(" \"value\":  \"NEVER\"")
+        .append(" }]");
+    logger.info("AdminServer shutdown patch string: {0}", patchStr);
+
+    patch = new V1Patch(new String(patchStr));
+    boolean crdPatched = assertDoesNotThrow(() ->
+            patchDomainCustomResource(domainUid, domainNamespace, patch, "application/json-patch+json"),
+        "patchDomainCustomResource(adminShutdown) failed");
+
+    assertTrue(crdPatched, "patchDomainCustomResource(adminShutdown) failed");
+   
+    checkPodDeleted(adminServerPodName, domainUid, domainNamespace);
+    logger.info("AdminServer shutdown success");
+
+    logger.info("Check managed server pods are not affected");
+    DateTime dynTs2 = getPodCreationTime(domainNamespace, dynamicServerPodName);
+    DateTime cfgTs2 = getPodCreationTime(domainNamespace, configServerPodName);
+
+    assertTrue(
+        dynTs2.withTimeAtStartOfDay().isEqual(dynTs.withTimeAtStartOfDay()), 
+        "Dynamic managed server pod creation time must be same");
+    assertTrue(
+        cfgTs2.withTimeAtStartOfDay().isEqual(cfgTs.withTimeAtStartOfDay()), 
+        "Configured managed server pod creation time must be same");
+
+    patchStr = null;
+    patchStr = new StringBuffer("[{");
+    patchStr.append("\"op\": \"replace\",")
+        .append(" \"path\": \"/spec/adminServer/serverStartPolicy\",")
+        .append(" \"value\":  \"IF_NEEDED\"")
+        .append(" }]");
+    logger.info("AdminServer restart patch string: {0}", patchStr);
+
+    patch = new V1Patch(new String(patchStr));
+    crdPatched = false;
+    crdPatched = assertDoesNotThrow(() ->
+            patchDomainCustomResource(domainUid, domainNamespace, patch, "application/json-patch+json"),
+        "patchDomainCustomResource(restart) failed");
+
+    assertTrue(crdPatched, "patchDomainCustomResource(restart) failed");
+    logger.info("Check admin service {0} is created in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkServiceExists(adminServerPodName, domainNamespace);
```

Comment:
use checkPodReadyAndServiceExists through out the class where ever it can be used

Reply:
done